.github/workflows/dark_storage_benchmark.yml

```diff
@@ -0,0 +1,78 @@
+name: Benchmark Dark Storage
+on:
+    schedule:
+        - cron: "0 0/3 * * *"
+    workflow_dispatch:
+
+permissions:
+    # deployments permission to deploy GitHub pages website
+    deployments: write
+    # contents permission to update benchmark contents in gh-pages branch
+    contents: write
+
+jobs:
+  check_for_new_commits:
+    runs-on: ubuntu-latest
+    steps:
+        - run: |
+              git clone https://github.com/equinor/ert.git
+              cd ert
+              latest_commit_date=$(git show -s --format=%ci)
+              current_date_3_hours_ago=$(date -d "3 hours ago" +"%Y-%m-%d %H:%M:%S %z")
+              commit_author=$(git show -s --format=%an)
+              if [[ "$latest_commit_date" > "$current_date_3_hours_ago" ]] && [[ "$commit_author" != "Function Key" ]];then
+                echo "git_new_commits=true" >> "$GITHUB_OUTPUT"
+                echo "Found new commits from the last 3 hours"
+                echo "Running benchmarks on git commit: $(git show -s --format=%H)"
+              else
+                echo "git_new_commits=false" >> "$GITHUB_OUTPUT"
+                echo "Found no new commits from the last 3 hours"
+              fi
+          id: check_for_commit
+    outputs:
+        git_new_commits: ${{steps.check_for_commit.outputs.git_new_commits}}
+
+  benchmark:
+    name: Run pytest-benchmark dark storage
+    needs: check_for_new_commits
+    if: needs.check_for_new_commits.outputs.git_new_commits == 'true'
+    timeout-minutes: 90
+    runs-on: ubuntu-latest
```

Comment:
As was discussed, do we want to run on onoprem instead?

Reply:
Yes, I will split it back into two workflows.